src/sinks/cratedb.rs

```diff
@@ -0,0 +1,556 @@
+use crate::{
+    buffers::Acker,
+    event::{unflatten::MapValue, Event},
+    sinks::util::{
+        http::{HttpRetryLogic, HttpService},
+        retries::FixedRetryPolicy,
+        Batch, BatchServiceSink, SinkExt,
+    },
+    topology::config::{DataType, SinkConfig},
+};
+use futures::{Future, Sink};
+use headers::HeaderMapExt;
+use http::Uri;
+use hyper::{Body, Client, Request};
+use hyper_tls::HttpsConnector;
+use serde::{Deserialize, Serialize, Serializer};
+use std::collections::HashMap;
+use std::time::Duration;
+use string_cache::DefaultAtom as Atom;
+use tower::ServiceBuilder;
+
+#[derive(Deserialize, Serialize, Clone, Debug, Default)]
+#[serde(deny_unknown_fields)]
+pub struct CrateDBSinkConfig {
+    pub uri: String,
+    pub schema: Option<String>,
```

Comment:
You can insert without a schema, at which point CrateDB will default to `doc`. I've used `Option` here because I have a default value: https://github.com/MarkusH/vector/blob/5a25571f9f1c869c2e4732b95aff757bcb4c0f6e/src/sinks/cratedb.rs#L144

But that comment made me realize I should probably document the table creation ...

Reply:
Ok that makes sense! 😄